client-device-connection-infinispan/src/main/java/org/eclipse/hono/deviceconnection/infinispan/client/HotrodCacheConfig.java

```diff
@@ -79,12 +83,15 @@ public RemoteCacheManager remoteCacheManager() {
      * in an Infinispan data grid.
      *
      * @param cache The remote cache in the Infinispan data grid.
+     * @param tracer The OpenTracing {@code Tracer} to use for tracking requests done by clients created by this factory.
+     *               If an empty Optional is given, the {@code NoopTracer} from OpenTracing will be used.
      * @return The factory.
      */
     @Bean
     @Qualifier(DeviceConnectionConstants.DEVICE_CONNECTION_ENDPOINT)
     @ConditionalOnProperty(prefix = "hono.device-connection", name = "server-list")
-    public BasicDeviceConnectionClientFactory hotrodBasedDeviceConnectionClientFactory(final HotrodCache<String, String> cache) {
-        return new HotrodBasedDeviceConnectionClientFactory(cache);
+    public BasicDeviceConnectionClientFactory hotrodBasedDeviceConnectionClientFactory(
+            final HotrodCache<String, String> cache, final Optional<Tracer> tracer) {
+        return new HotrodBasedDeviceConnectionClientFactory(cache, tracer.orElse(NoopTracerFactory.create()));
```

Comment:
this will throw a `NullPointerException` if *tracer* is `null`, right? Is there actually a factory method producing an `Optional<Tracer>`?

Reply:
The method is supposed to be called by Spring.
Using `Optional` here is a way to tell Spring that such a `Tracer` bean is optional. In that sense it is an equivalent to using `@Autowired(required = false)` on a field or method (using Autowired on a method parameter is mostly ignored, as written [here](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/factory/annotation/Autowired.html). Another alternative would be to use a `@Nullable` annotation, which I also see as a good solution, which would require an additional dependency though (`com.google.code.findbugs:jsr305` I guess we haven't discussed usage of that yet).

Reply:
Ah, I learn something new every day :-) In any case, IMHO you should document the `NullPointerException` ...